src/Process/Deploy/CheckEnvVarMageErrorReportDirNestingLevel.php

```diff
@@ -0,0 +1,75 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+namespace Magento\MagentoCloud\Process\Deploy;
+
+use Magento\MagentoCloud\Config\Environment;
+use Magento\MagentoCloud\Filesystem\ConfigFileList;
+use Magento\MagentoCloud\Process\ProcessException;
+use Magento\MagentoCloud\Process\ProcessInterface;
+use Psr\Log\LoggerInterface;
+
+/**
+ * @inheritdoc
+ */
+class CheckEnvVarMageErrorReportDirNestingLevel implements ProcessInterface
+{
+    const ENV_MAGE_ERROR_REPORT_DIR_NESTING_LEVEL = 'MAGE_ERROR_REPORT_DIR_NESTING_LEVEL';
+
+    /**
+     * @var LoggerInterface
+     */
+    private $logger;
+
+    /**
+     * @var Environment
+     */
+    private $environment;
+
+    /**
+     * @var ConfigFileList
+     */
+    private $configFileList;
+
+    /**
+     * @param LoggerInterface $logger
+     * @param Environment $environment
+     * @param ConfigFileList $configFileList
+     */
+    public function __construct(
+        LoggerInterface $logger,
+        Environment $environment,
+        ConfigFileList $configFileList
+    ) {
+        $this->logger = $logger;
+        $this->environment = $environment;
+        $this->configFileList = $configFileList;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function execute()
+    {
+        try {
+            $envReportDirNestingLevel = $this->environment->getEnv(self::ENV_MAGE_ERROR_REPORT_DIR_NESTING_LEVEL);
+            if ($envReportDirNestingLevel) {
+                $this->logger->notice(
+                    sprintf(
+                        '%s environment variable detected. %s: %s. '
+                        .'Value of the property \'config.report.dir_nesting_level\' from \'%s\' file be ignored',
+                        self::ENV_MAGE_ERROR_REPORT_DIR_NESTING_LEVEL,
+                        self::ENV_MAGE_ERROR_REPORT_DIR_NESTING_LEVEL,
+                        $envReportDirNestingLevel,
+                        $this->configFileList->getErrorReportConfig()
+                    )
```

Comment:
First, I would write this message only if customer define value for this variable in `env.yaml` file.
Also, this notice message duplicate the mane of variable, please rephrase it. E.g.
`Environment variable MAGE_ERROR_REPORT_DIR_NESTING_LEVEL exists with value {value}. Value of ERROR_REPORT_DIR_NESTING_LEVEL  defined in .magento.app.yaml will be ignored.`

Reply:
Changed